### PR TITLE
Further updates to weather selection/config

### PR DESCRIPTION
**Pull request by Skipper-is**
Further updates:
Weather config added to main options window
image
Saves the info to the .flight.cfg file in the same wording as the .ysf:
```
CONSTWIND -2.244669m/s 0.000000m/s -4.049489m/s
CLOUDLAYER OVC 60.960000m 152.400000m
CLOUDLAYER OVC 1158.239998m 1249.679998m
```

Have amended language to reflect the units being used (ft and NM). 
Config defaults now propagate through to the create new flight menu. 

@pasutisu - could you review the Japanese wording to ensure it makes sense?

### CODE REVIEW

**Review by NodokaHanamura — APPROVED** [review]

**Comment by JOBBIN9422** [comment]
tested (briefly) on my end - looks good!

File: src/core/fsguiconfigdlg.cpp [hunk 1]
```diff
@@ -439,6 +467,21 @@ void FsGuiConfigDialog::InitializeDialog(FsWorld *,FsFlightConfig &cfg)
 		dayOrNightBtn[1]->SetCheck(YSTRUE);
 		break;
 	}
+	double windDirval = -atan2(cfg.constWind.x(), -cfg.constWind.z()) * 180.0 / YsPi;
```
Comment (JOBBIN9422) [inline src/core/fsguiconfigdlg.cpp:470]: consider using YsRadToDeg() method here for the conversion (not a big deal)

File: src/core/fsguiconfigdlg.cpp [hunk 2]
```diff
@@ -587,7 +630,27 @@ void FsGuiConfigDialog::RetrieveConfig(FsFlightConfig &cfg)
 		cfg.env=FSNIGHT;
 	}
 
+	double windDirVal = windDir->GetNumber()/(180/YsPi); //In rads
```
Comment (JOBBIN9422) [inline src/core/fsguiconfigdlg.cpp:633]: consider using YsDegToRad() for conversion here

File: src/core/fssiminfo.cpp [hunk 3]
```diff
@@ -1802,6 +1802,13 @@ void FsNewFlightDialogInfo::SetFromConfig(const class FsWorld *world,const class
 	fieldName.Set(cfg.defField);
 	playerAirInfo.typeName.Set(cfg.defAirplane);
 	playerAirInfo.startPos.Set(cfg.defStartPos);
+	//Set the environmental data from the defaults
+	envInfo.windDir = -atan2(cfg.constWind.x(), -cfg.constWind.z()) * 180.0 / YsPi;
```
Comment (JOBBIN9422) [inline src/core/fssiminfo.cpp:1806]: consider using YsRadToDeg() here

**Review by JOBBIN9422 — APPROVED** [review]
approved with minor comments (rad <-> deg conversions)

**Review by pasutisu — APPROVED** [review]

**Pull request merged**